### ISSUE TITLE
Docker based setup for self hosting

### ISSUE THREAD

**Issue by accforgithubtest** [nkanaev/yarr#173]
Hello there

A docker compose based set up might be the most ideal option for selfhosting. Would be great if there is a 
1. Image on dockherhub / etc, prebuilt for all platforms.
2. Detailed instructions with a sample example file.
3. Provide a single config / file that can be volume mounted for configuring the list of RSS feeds.
4. Environment variables for configuring refresh frequency, etc.

Thank you for this project!

**Comment by grabpot** [comment]
If it helps, here's my recent experience on this topic:
* There is already an unofficial (?) image on dockerhub: https://hub.docker.com/r/arsfeld/yarr
* The default `etc/dockerfile` runs `make build_linux` which assumes it's compiling for amd64. If you're using an arm64 device, e.g. a Raspberry Pi, then you'll need to run `make build_default` instead.
* There's currently an issue with compiling go-sqlite3 on the latest Go 1.21.5 docker image (used by the above Dockerfile). Adding `-D_LARGEFILE64_SOURCE` to the `CGO_CFLAGS` environment variable is a temporary fix. For further information, see: https://github.com/mattn/go-sqlite3/pull/1177
* You can already import/export an OPML file with the list of RSS feeds from the web interface, if that helps.

@nkanaev I'm happy to raise a PR to add this to the docs if you like.

I ended up with the following Dockerfile:
```
FROM arm64v8/golang:1.21.5-alpine AS build
RUN apk add --no-cache build-base git

WORKDIR /src
RUN git clone https://github.com/nkanaev/yarr.git .
ENV CGO_CFLAGS="-D_LARGEFILE64_SOURCE"
RUN make build_default

FROM alpine:3.19
COPY --from=build /src/_output/yarr /usr/local/bin/yarr
RUN mkdir /data
CMD ["/usr/local/bin/yarr", "-addr", "0.0.0.0:7070", "-db", "/data/yarr.db"]
```

I'm running yarr behind Caddy with Tailscale HTTPS, so my `docker-compose.yml` file is a bit more complicated, but a basic example might look like this, assuming it's in the same directory as the Dockerfile and that you've also created a sub-directory called `data`:
```
version: "3"

services:
  yarr:
    build:
      context: .
    ports:
      - 7070:7070
    volumes:
      - ./data:/data
```

**Comment by rebron1900** [comment]
> 如果有帮助的话，这是我最近在这个主题上的经验：
> 
> * dockerhub 上已经有一个非官方（？）镜像：https://hub.docker.com/r/arsfeld/yarr
> * 默认的 `etc/dockerfile` 运行 `make build_linux` ，它假定它是针对 amd64 进行编译的。如果您使用的是arm64设备，例如Raspberry Pi，那么您需要运行 `make build_default` 。
> * 目前在最新的 Go 1.21.5 Alpine docker 镜像（上面的 Dockerfile 使用）上编译 go-sqlite3 存在问题。将 `-D_LARGEFILE64_SOURCE` 添加到 `CGO_CFLAGS` 环境变量是一个临时修复。有关更多信息，请参阅：修复 musl 构建 (#1164) mattn/go-sqlite3#1177
> * 如果有帮助的话，您已经可以从 Web 界面导入/导出包含 RSS 源列表的 OPML 文件。
> 
> @nkanaev 如果您愿意，我很乐意提出 PR 将其添加到文档中。
> 
> 我最终得到了以下 Dockerfile：
> 
> ```
> FROM golang:1.21.5-alpine AS build
> RUN apk add --no-cache build-base git
> 
> WORKDIR /src
> RUN git clone https://github.com/nkanaev/yarr.git .
> ENV CGO_CFLAGS="-D_LARGEFILE64_SOURCE"
> RUN make build_default
> 
> FROM alpine:3.19
> COPY --from=build /src/_output/yarr /usr/local/bin/yarr
> RUN mkdir /data
> CMD ["/usr/local/bin/yarr", "-addr", "0.0.0.0:7070", "-db", "/data/yarr.db"]
> ```
> 
> 我使用 Tailscale HTTPS 在 Caddy 后面运行 yarr，因此我的 `docker-compose.yml` 文件有点复杂，但基本示例可能如下所示，假设它与 Dockerfile 位于同一目录中，并且您已经还创建了一个名为 `data` 的子目录：
> 
> ```
> version: "3"
> 
> services:
>   yarr:
>     build:
>       context: .
>     ports:
>       - 7070:7070
>     volumes:
>       - ./data:/data
> ```

Hello, I forked a repository called [rebron1900/yarr](https://github.com/rebron1900/yarr) and made some small improvements. 

I want to publish it to Docker Hub using Action, but I encountered an error while running it. After my investigation, I found that the master branch and tag branch of the original author have different file structures. How did they manage to do that?

How can I make it run properly?


> If it helps, here's my recent experience on this topic:
> 
> * There is already an unofficial (?) image on dockerhub: https://hub.docker.com/r/arsfeld/yarr
> * The default `etc/dockerfile` runs `make build_linux` which assumes it's compiling for amd64. If you're using an arm64 device, e.g. a Raspberry Pi, then you'll need to run `make build_default` instead.
> * There's currently an issue with compiling go-sqlite3 on the latest Go 1.21.5 Alpine docker image (used by the above Dockerfile). Adding `-D_LARGEFILE64_SOURCE` to the `CGO_CFLAGS` environment variable is a temporary fix. For further information, see: [Fix musl build (#1164) mattn/go-sqlite3#1177](https://github.com/mattn/go-sqlite3/pull/1177)
> * You can already import/export an OPML file with the list of RSS feeds from the web interface, if that helps.
> 
> @nkanaev I'm happy to raise a PR to add this to the docs if you like.
> 
> I ended up with the following Dockerfile:
> 
> ```
> FROM golang:1.21.5-alpine AS build
> RUN apk add --no-cache build-base git
> 
> WORKDIR /src
> RUN git clone https://github.com/nkanaev/yarr.git .
> ENV CGO_CFLAGS="-D_LARGEFILE64_SOURCE"
> RUN make build_default
> 
> FROM alpine:3.19
> COPY --from=build /src/_output/yarr /usr/local/bin/yarr
> RUN mkdir /data
> CMD ["/usr/local/bin/yarr", "-addr", "0.0.0.0:7070", "-db", "/data/yarr.db"]
> ```
> 
> I'm running yarr behind Caddy with Tailscale HTTPS, so my `docker-compose.yml` file is a bit more complicated, but a basic example might look like this, assuming it's in the same directory as the Dockerfile and that you've also created a sub-directory called `data`:
> 
> ```
> version: "3"
> 
> services:
>   yarr:
>     build:
>       context: .
>     ports:
>       - 7070:7070
>     volumes:
>       - ./data:/data
> ```


Hello, I forked a repository called [rebron1900/yarr](https://github.com/rebron1900/yarr) and made some small improvements. 

I want to publish it to Docker Hub using Action, but I encountered an error while running it. After my investigation, I found that the master branch and tag branch of the original author have different file structures. How did they manage to do that?

How can I make it run properly?

**Comment by rebron1900** [comment]
@grabpot thks，I have successfully packaged my own Docker image.

**Comment by darkdragon-001** [comment]
Would be great if we could add a GitHub Actions workflow to publish the image automatically on releases.

Here are the [docs](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images).

**Comment by darkdragon-001** [comment]
@rebron1900 I just saw your workflow at https://github.com/rebron1900/yarr/blob/master/.github/workflows/build-docker-image. Maybe do you want to create a PR to this repository?

**Comment by rebron1900** [comment]
> @rebron1900 I just saw your workflow at https://github.com/rebron1900/yarr/blob/master/.github/workflows/build-docker-image. Maybe do you want to create a PR to this repository?

Yes, this workflow works really well. And I've also added support for one -click subscription to yarr through Rsshub radar.
Of course, I'm willing to do that. I will try to submit a pull request for the master branch later.
